.github/workflows/therock-ci-linux.yml

```
@@ -0,0 +1,136 @@
+name: TheRock CI Linux
+
+on:
+  workflow_call:
+    inputs:
+      cmake_options:
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  therock-build-linux:
+    name: Build Linux Packages
+    runs-on: azure-linux-scale-rocm
+    permissions:
+      id-token: write
+    container:
+      image: ghcr.io/rocm/therock_build_manylinux_x86_64@sha256:044b113562629f4bd2ec5d2e64b32eee11562d48fb1a75d7493daec9dd8d8292
+    env:
+      CACHE_DIR: ${{ github.workspace }}/.container-cache
+      CCACHE_DIR: "${{ github.workspace }}/.container-cache/ccache"
+      CCACHE_MAXSIZE: "700M"
+      AMDGPU_FAMILIES: "gfx94X-dcgpu"
+      TEATIME_FORCE_INTERACTIVE: 0
+    steps:
+      - name: Checkout composable_kernel repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Checkout TheRock repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: "ROCm/TheRock"
+          ref: ec1c2ef4f2636bce7733fd8c95e1dbb6692c8a57
+          path: "TheRock"
+
+      - name: Runner Health Settings
+        run: |
+          echo "CCACHE_DIR=${CCACHE_DIR}"
+          df -h
+          ccache -z
+          mkdir -p $CCACHE_DIR
+          cmake --version
+          echo "Installed Python versions:"
+          ls -d /opt/python
+          echo "python: $(which python), python3: $(which python3)"
+          echo "Git version: $(git --version)"
+          git config --global --add safe.directory $PWD
+          git config fetch.parallel 10
+      
+      - name: Fetch sources
+        run: |
+          ./TheRock/build_tools/fetch_sources.py --jobs 12 --include-rocm-libraries
+
+      - name: Install python deps
+        run: |
+          pip install -r TheRock/requirements.txt
+          pip freeze
+
+      - name: Configure Projects
+        env:
+          amdgpu_families: ${{ env.AMDGPU_FAMILIES }}
+          package_version: ADHOCBUILD
+          extra_cmake_options: "-DTHEROCK_USE_EXTERNAL_CK=ON -DTHEROCK_CK_SOURCE_DIR=../ ${{ inputs.cmake_options }}"
```

```suggestion
          extra_cmake_options: "${{ inputs.cmake_options }} -DTHEROCK_USE_EXTERNAL_CK=ON -DTHEROCK_CK_SOURCE_DIR=../"
```
Might be easier to read, wdyt?

---

Furthermore, should `THEROCK_USE_EXTERNAL_CK` and `-DTHEROCK_CK_SOURCE_DIR` eventually be moved to `inputs.cmake_options`? With this it seems the workflow could be nearly reusable by other projects as well.

---

We're going to have a lot of these options. Can we standardize the naming. 

THEROCK_EXTERNAL_(REPONAME)_ENABLE
THEROCK_EXTERNAL_(REPONAME)_SOURCE_DIR

Make sure the REPONAME matches was it is called in .gitmodules

---

This would require changes in TheRock which I am happily taking, allowing @geomin12 to pick it up here later.

---

that works! I pinned a commit hash for TheRock, so this can be landed and update in the future!